src/code/sys_cfb.c

```diff
@@ -1,11 +1,11 @@
 #include "global.h"
 
-u32 sSysCfbFbPtr[2];
-u32 sSysCfbEnd;
+uintptr_t sSysCfbFbPtr[2];
+uintptr_t sSysCfbEnd;
```

Comment:
Can these be changed to pointer types instead?

Reply:
I tried to do that but I couldn't get anything satisfying and it had a few issues, unfortunately.

`sSysCfbEnd` doesn't work well as a pointer because of the operations done in `SysCfb_Init`, not without a ton of casts at least

`sSysCfbFbPtr` could theoretically be a pointer with a few extra casts, but it wouldn't make much sense to use pointers for one and not the other, and it doesn't feel better overall tbh

Reply:
I think at the very least the function return types should be pointer types, in general it feels very wrong to rely on intptr types where pointer types are usable.
I agree that `sSysCfbEnd` itself being typed as a pointer would introduce a great deal of casts due to the masking and arithmetic done with/on it but as far as the external interface is concerned it should be treated as a pointer. I don't think some extra casts in sys_cfb outweigh this.

I think `debugHeapStart` in `main.c` also stands to gain from being typed as a pointer, yes it will need a cast for `PHYS_TO_K0(0x600000) - debugHeapStart` but highlighting this line as "weird" is more useful than casting between uintptr_t and void* elsewhere in the function

Reply:
Pushed a commit that changes the `sys_cfb` functions to return pointer types and makes `debugHeapStart` a pointer in `main.c`